contributing/testing-best-practices.md

```diff
@@ -0,0 +1,61 @@
+- `internal/testutils/mig` contains helper test functions specifically for Migration tests.
+- `internal/testutils/replay` contains helper test functions for [Hoverfly](https://docs.hoverfly.io/en/latest/). Hoverfly is used to capture and replay HTTP traffic with Atlas Cloud.
+
+## Unit tests
+
+- Unit tests must not create Terraform resources or use external systems like [Atlas Go SDK](https://github.com/mongodb/atlas-sdk-go).
+- Mock of specific interfaces like `client.ProjectsApi` is prefered to use the whole `client`.
+- [Testify Mock](https://pkg.go.dev/github.com/stretchr/testify/mock) is used for test doubles.
+- Altlas Go SDK mocked interfaces are generated in [mockadmin](https://github.com/mongodb/atlas-sdk-go/tree/main/mockadmin) package using [Mockery](https://github.com/vektra/mockery).
+
+## Acceptance tests
+
+- There must be at least one `basic acceptance test` for each resource.
+- `Basic import tests` are done as the last step in the `basic acceptance tests`, not as a different test. Exceptions apply for more specific import tests, e.g. testing with incorrect IDs. [Import tests](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/import#resource-acceptance-testing-implementation) verify that the [Terraform Import](https://developer.hashicorp.com/terraform/cli/import) functionality is working fine.
+- Data sources are tested in the same tests as the resources. There are no separate files or tests for data sources as a resource must typically be created. (There are very few exceptions to this, e.g. when there is only data sources but not resource)
+- Helper functions such as `resource.TestCheckTypeSetElemNestedAttrs` or `resource.TestCheckTypeSetElemAttr` can be used to check resource and data source attributes more easily.
+
+## Migration tests
+
+- There must be at least one `basic migration test` for each resource that leverages on the `basic acceptance tests` using helper test functions such as `CreateAndRunTest`.
+
+## Local development
+
+These enviroment variables can be used in local to speed up development process.
+
+Enviroment Variable | Description
+--- | ---
+`MONGODB_ATLAS_PROJECT_ID` | Re-use an existing project reducing test run duration for resources supporting this variable
+`MONGODB_ATLAS_CLUSTER_NAME` | Re-use an existing cluster reducing significantly test run duration for resources supporting this variable
+`REPLAY_MODE` | Use [Hoverfly](https://docs.hoverfly.io/en/latest/), more info about possible variable values [here](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/development-setup.md#replaying-http-requests-with-hoverfly)
+
+## Shared resources
+
+Acceptance and migration tests can reuse projects and clusters in order to be more efficient in resource utilization.
+
+- A project can be reused using `ProjectIDExecution`. It returns the ID of a project that is created for the current execution of tests for a resource.
```

Comment:
reading usage seems to be 
```suggestion
- A project can be reused using `acc.ProjectIDExecution(t)`. It returns the ID of a project that is created for the current execution of tests for a resource.
```
But I would suggest this may be better explained with a code block that includes the imports and comments on all the bits needed,